### PR TITLE
[SPARK-31907][DOCS][SQL] Adding location of SQL API documentation

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
Linking to location of SQL API documentation, making it easier and quicker to find it.

### Why are the changes needed?
documentation clarity

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Only documentation change

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ import org.apache.spark.util.Utils
  * Spark also includes more built-in functions that are less common and are not defined here.
  * You can still access them (and all the functions defined here) using the `functions.expr()` API
  * and calling them through a SQL expression string. You can find the entire list of functions
- * at SQL API documentation.
+ * at SQL API documentation (https://spark.apache.org/docs/latest/api/sql/index.html).
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:48]: the problem is that the link became stale. For example, in Spark 3.0.0 release documentation, https://spark.apache.org/docs/latest/api/sql/index.html will point out the latest version of Spark.

Comment (dominikgehl) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: I tried to address this issue with an additional commit.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -45,7 +45,8 @@ import org.apache.spark.util.Utils
  * Spark also includes more built-in functions that are less common and are not defined here.
  * You can still access them (and all the functions defined here) using the `functions.expr()` API
  * and calling them through a SQL expression string. You can find the entire list of functions
- * at SQL API documentation.
+ * at SQL API documentation (https://spark.apache.org/docs/latest/api/sql/index.html or
+ * https://spark.apache.org/docs/<version>/api/sql/index.html for older versions).
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:49]: How about his?

```suggestion
 * at SQL API documentation of your Spark version, see also
 * <a href="https://spark.apache.org/docs/latest/api/sql/index.html">the latest list</a>
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**